cli/azd/test/mocks/util.go

```diff
@@ -31,3 +31,22 @@ func CreateEmptyHttpResponse(request *http.Request, statusCode int) (*http.Respo
 		Body:       http.NoBody,
 	}, nil
 }
+
+func RemoveArg(arg string, args []string) ([]string, string, string) {
+	result := make([]string, len(args)-2)
+	var insertIndex int
+	var extractArg, value string
+	for index := range args {
+		if args[index] == arg {
+			extractArg = args[index]
+			continue
+		}
+		if index > 0 && args[index-1] == arg {
+			value = args[index]
+			continue
+		}
+		result[insertIndex] = args[index]
+		insertIndex++
+	}
+	return result, extractArg, value
+}
```

Comment:
It's not very clear how and when someone would use this. Consider adding some inline docs.  So we're removing an argument from an array of arguments returning the new list of args, the extracted arg and its value?

Reply:
did not care too much about documenting this as it is for making test cases to ignore the `iidfile` arg, as it changes on every run

Reply:
I think it would be worthwhile to add a small doc comment here anyway, so folks have a better handle on how to hold this thing (or to rework the tests so it's not needed).

Something as small as:

```
// RemoveArg takes an argument name and an slice of arguments and returns three values:
 -  a new slice, which is a copy of the input argument slice, except without the named argument and its value.
 - The name of the argument.
 - The value of the argument.
```

Typing this all out makes me realize that the second return value is sort of useless, since it will always be the same as the first argument to this function, and indeed looking at the code we always discard this value.

Perhaps it would be better to write something like:

`ExtractArg(arg string, args []string) (value string, newArgs []string)`, where `value` is the value of `arg` in args and `newArgs` is just `args` but with the `arg` and it's corresponding value removed from the array?